webapp/tests/elster_client/mock_erica.py

```diff
@@ -345,3 +356,29 @@ def errors_from_error_flags(show_response):
         # InvalidTaxNumberError
         if MockErica.invalid_tax_number_error_occurred:
             return get_json_response('invalid_tax_number')
+
+    request_id_count = {}
+
+    @staticmethod
+    def post_dummy_job(*args, **kwargs):
+        if Config.MOCK_DELIVER_FAIL_ON_POST:
+            return {"errorCode": -1, "errorMessage": "Job could not be submitted."}, 422
+        else:
+            request_id = str(uuid.uuid4())
+            MockErica.request_id_count.setdefault(request_id, 0)
+            return "/02/get_job/" + request_id, 201
+
+    @staticmethod
+    def get_dummy_job(request_id):
+        count = MockErica.request_id_count.get(request_id, -1) + 1
+        if count >= Config.MOCK_MIN_REQUEST_COUNT:
+            if Config.MOCK_DELIVER_FAIL_ON_GET:
+                response = {"processStatus": "failure", "errorCode": -1, "errorMessage": "ELSTER Timeout error"}
+            else:
+                response = {"processStatus": "success", "payload": get_json_response('est_without_responses')}
+        elif count == 0:
+            response = {"errorCode": -1, "errorMessage": "Request ID not found"}, 500
```

Comment:
Wouldn't that be a 404?

Reply:
I just followed the tech spec. There you only defined a 422 for errors, but yes we should use 404 for missing request id. So 422 is ok here?

Reply:
Changed 500 error to 404 for request id not found